specifyweb/specify/record_merging.py

```diff
@@ -221,10 +204,20 @@ def record_merge_fx(model_name: str, old_model_ids: List[int], new_model_id: int
             field_name = col.lower()
             field_name_id = f'{field_name}_id'
             if not hasattr(foreign_model, field_name_id):
-                if (field_name.endswith('s') and hasattr(foreign_model, field_name)) \
-                    and foreign_table.get_field(field_name).type.endswith('to-many'):
-                    field_name_id = field_name
-                else:
+                continue
+
+
+            # Handle case of updating a large amount of record ids in a foreign table.
+            # Example: handle case of updating a large amount of agent ids in the audit logs.
+            # Fix by optimizing the query by consolidating it here
+            if model_name.lower() in MERGING_OPTIMIZATION_FIELDS and \
+                table_name.lower() in MERGING_OPTIMIZATION_FIELDS[model_name.lower()]:
+                if field_name_id in MERGING_OPTIMIZATION_FIELDS[model_name.lower()][table_name.lower()]:
+                    query = Q(**{field_name_id: old_model_ids[0]})
+                    for old_model_id in old_model_ids[1:]:
+                        query.add(Q(**{field_name_id: old_model_id}), Q.OR)
+                    foreign_model.objects.filter(query).update(**{field_name_id: new_model_id})
+                    progress(1, 0) if progress is not None else None
```

Comment:
@acwhite211  see https://github.com/specify/specify7/pull/4699, on why this was moved.